### PR TITLE
API update

### PR DESCRIPTION
**Pull request by tasdomas**
Update API paths (env -> model)
Add PATCH methods.
Add content type support.

### CODE REVIEW

**Comment by alesstimec** [comment]
LGTM!

File: api/budget/api.go [hunk 1]
```diff
@@ -138,6 +138,12 @@ type hasMethod interface {
 	Method() string
 }
 
+// hasContentType is an interface implemented by reqyests to
```
Comment (mattyw) [inline api/budget/api.go]: requests

**Comment by mattyw** [comment]
one spelling mistake

**Comment by tasdomas** [comment]
@alesstimec, @mattyw - PTAL

**Comment by tasdomas** [comment]
$$merge$$

**Pull request merged**